api/resolvers/item.js

```diff
@@ -1066,21 +1075,39 @@ export const createItem = async (parent, { forward, options, ...item }, { me, mo
   }
 
   const enforceFee = me ? undefined : (item.parentId ? ANON_COMMENT_FEE : (ANON_POST_FEE + (item.boost || 0)))
+  const deleteCommand = getDeleteCommand(item.text)
+  const isEphemeral = !!deleteCommand
+
   item = await serializeInvoicable(
     models.$queryRawUnsafe(
-      `${SELECT} FROM create_item($1::JSONB, $2::JSONB, $3::JSONB, '${spamInterval}'::INTERVAL) AS "Item"`,
+      `${SELECT} FROM create_item($1::JSONB, $2::JSONB, $3::JSONB, '${spamInterval}'::INTERVAL, ${isEphemeral ? EPHEMERAL_FEE_MSATS : 0}::BIGINT) AS "Item"`,
       JSON.stringify(item), JSON.stringify(fwdUsers), JSON.stringify(options)),
     { models, lnd, hash, hmac, me, enforceFee }
   )
 
   await createMentions(item, models)
 
+  await enqueueDeletionJob(item, models, deleteCommand)
+
   notifyUserSubscribers({ models, item })
 
   item.comments = []
   return item
 }
 
+const clearDeletionJobs = async (item, models) => {
+  await models.$queryRawUnsafe(`DELETE FROM pgboss.job WHERE name = 'deleteItem' AND data->>'id' = '${item.id}';`)
+}
+
+const enqueueDeletionJob = async (item, models, incomingCommand) => {
+  const deleteCommand = incomingCommand ?? getDeleteCommand(item.text)
+  if (deleteCommand) {
+    await models.$queryRawUnsafe(`
+      INSERT INTO pgboss.job (name, data, startafter)
+      VALUES ('deleteItem', jsonb_build_object('id', ${item.id}), now() + interval '${deleteCommand.number} ${deleteCommand.unit}s');`)
+  }
+}
+
```

Comment:
@huumn would you rather me try to incorporate the pgboss job sql queries into the `create_item` and `update_item` SQL functions, so they are part of the same serialized transaction, or can they be done separate from those function calls, similar to how we do `createMentions` now? That'll dictate the direction I take this PR a bit, so I wanted to get your feedback on that

Reply:
Let's go with a mention-style approach for now. Those postgres functions are getting bloated.